core/src/main/scala/org/apache/spark/ui/jobs/JobPage.scala

```diff
@@ -32,7 +32,12 @@ private[ui] class JobPage(parent: JobsTab) extends WebUIPage("job") {
 
   def render(request: HttpServletRequest): Seq[Node] = {
     listener.synchronized {
-      val jobId = request.getParameter("id").toInt
+      val parameterId = request.getParameter("id")
```

Comment:
Elsewhere the code uses `require(condition, message)` to generate an `IllegalArgumentException`. I think that could be tidier here. You can also use `isEmpty`:
`require(parameterId != null && !parameterId.isEmpty, "Missing id parameter")`